base_layer/core/src/transactions/aggregated_body.rs

```diff
@@ -117,11 +118,48 @@ impl AggregateBody {
         self.kernels.push(kernel);
     }
 
+    /// Add a kernels to the existing aggregate body
+    pub fn add_kernels(&mut self, kernels: &mut Vec<TransactionKernel>) {
+        self.kernels.append(kernels);
+        self.sorted = false;
+    }
+
     /// Set the kernel of the aggregate body, replacing any previous kernels
     pub fn set_kernel(&mut self, kernel: TransactionKernel) {
         self.kernels = vec![kernel];
     }
 
+    /// This will perform cut-through on the aggregate body
+    pub fn do_cut_through(&mut self) {
+        let mut remove_hash = Vec::new();
```

Comment:
try something like
```
self.outputs.iter().filter(|output| 
    self.inputs.iter().map(|i| i.hash()).contains(output.hash())
).collect()
```

Reply:
Also, Adding a `is_equal_to(utxo: TransactionOutput)` method to `TransactionInput` could probably save us a ton of hashes by directly comparing the UTXO commitment and feature set. Then the above becomes 

```
self.outputs.iter().filter(|output| 
    self.inputs.iter().any(|i| i.is_equal_to(output)
).collect()
```